common/Resources/ti.internal/extensions/util.js

```diff
@@ -0,0 +1,498 @@
+			// now look at next char to see how to replace
+			const nextChar = firstArg.charAt(foundIndex + 1);
+			switch (nextChar) {
+				case 's': // string
+					str += String(curArg);
+					i++; // consume argument
+					break;
+
+				case 'd': // Number
+					if (util.isSymbol(curArg) || util.types.isSymbolObject(curArg)) {
+						str += 'NaN';
+					} else {
+						str += Number(curArg);
+					}
+					i++; // consume argument
+					break;
+
+				case 'i': // Integer
+					if (util.isSymbol(curArg) || util.types.isSymbolObject(curArg)) {
+						str += 'NaN';
+					} else {
+						str += parseInt(curArg);
+					}
+					i++; // consume argument
+					break;
+
+				case 'f': // Float
+					if (util.isSymbol(curArg) || util.types.isSymbolObject(curArg)) {
+						str += 'NaN';
+					} else {
+						str += parseFloat(curArg);
+					}
+					i++; // consume argument
+					break;
+
+				case 'j':  // JSON
+					str += stringify(curArg);
+					i++; // consume argument
+					break;
+
+				case 'o': // Object w/showHidden and showProxy
+					str += util.inspect(curArg, { showHidden: true, showProxy: true, depth: 4 });
+					i++; // consume argument
+					break;
+
+				case 'O': // Object w/o options
+					str += util.inspect(curArg, {});
+					i++; // consume argument
+					break;
+
+				case '%': // escaped %
+					str += '%';
+					// Don't consume argument here!
+					break;
+			}
+			lastIndex = foundIndex + 2;
+		}
+
+		// If we haven't reached end of string, append rest of it with no replacements!
+		str += firstArg.slice(lastIndex, firstArg.length);
+
+		// If we have args remaining, need to...
+		// loop over rest of args and coerce to Strings and concat joined by spaces.
+		// FIXME: Unless typeof === 'object' or 'symbol', then do util.inspect() on them
+		if (i < args.length) {
+			str += ` ${args.slice(i).map(a => String(a)).join(' ')}`;
+		}
+		return str;
+	}
+
+	// first arg wasn't string, so we loop over args and call util.inspect on each
+	return args.map(a => util.inspect(a)).join(' ');
+};
+
+/**
+ * @param {Function} constructor subclass
+ * @param {Function} superConstructor base class
+ * @returns {void}
+ */
+util.inherits = function (constructor, superConstructor) {
+	// TODO Extract common code for enforcing types of arguments. We use this logic in path.js too!
+	if (constructor == null) { // eslint-disable-line eqeqeq, no-eq-null
+		throw new TypeError(`The "constructor" argument must be of type Function. Received type ${typeof constructor}`);
+	}
+	if (superConstructor == null) { // eslint-disable-line eqeqeq, no-eq-null
+		throw new TypeError(`The "superConstructor" argument must be of type Function. Received type ${typeof superConstructor}`);
+	}
+	if (superConstructor.prototype === undefined) {
+		throw new TypeError('The "superConstructor.prototype" argument must be of type Object. Received type undefined');
+	}
+
+	constructor.super_ = superConstructor; // FIXME: Don't make enumerable!
```

Comment:
Can't this be fixed with `Object.defineProperty`?
```suggestion
	Object.defineProperty(constructor, 'super_', superConstructor);
```

Reply:
yup, I'll push a fix.